### DISCUSSION TITLE
Displacement boundaries update problem

### DISCUSSION THREAD

**Discussion by muhammedfurkanyilmaz** [KratosMultiphysics/Kratos#12277] [Q&A]
Dear @Kratos/Geomechanics,

I want to make a two-stage analysis by changing the displacement boundaries and loading process. At the first stage I've just tried to define K0 conditions with the lateral displacement boundaries normally fixed, and at the second stage, I've added absorbing boundaries at the base and allowed to displace horizontally at the lateral boundaries. However, the displacement boundaries of the first stage were not updated in the second stage. What could be the reason?

Best regards,
Furkan

**Comment by rfaasse** [comment]
Hi @muhammedfurkanyilmaz,
Thank you for reaching out! To tag the team, could you please use @KratosMultiphysics/geomechanics instead of @Kratos/Geomechanics? That way we'll all get the notification! I hope we can get back to you soon with an answer on this particular question. Thanks again for reaching out!
Kind regards,
Richard

**Comment by WPK4FEM** [comment]
Dear Furkan,
What I understand from your description is that the sides remain fixed in the second stage. I could guess that the "processes_sub_model_part_list" in the ProjectParameters.json file for the second stage still contains the modelpart for the side fixes, but without your model files I can not be sure. Please attach your .mdpa and .json files such that we can check and provide you a verified answer.
Kind regards,
Wijtze Pieter

**Reply by muhammedfurkanyilmaz** [reply]
Dear Wijtze,
Related files are attached. 
[LysmerBoundaries-K0.zip](https://github.com/KratosMultiphysics/Kratos/files/15215518/LysmerBoundaries-K0.zip)

Kind Regards
Furkan

**Comment by WPK4FEM** [comment]
Dear @muhammedfurkanyilmaz,

Inspecting your files, the following things I looked at
Changes from stage 1 to stage 2:
- Rayleigh parameters
- boundary condition side -> side2, switching from X support to Y support on same nodes.
- vertical volume acceleration disappears ( stress release prevented partly by Y supports on sides), I would leave the vertical acceleration in place as the  the middle nodes of the column are free in Y direction

Other things that stand out:
- stage 1 has "Lower_Drainage_Fluid_Pressure","Fluid_Pressure-Initial_saturated" in the projectparameters, but they are not used. stage 1 does not seem to have a boundary condition, nor an initial field for the water pressure.


To get a for me understandable starting situation as result of stage 1, I changed the following:
- added initial field for the water pressure
- added boundary condition for the water pressure at the bottom
- made a stage 1 material with IGNORE_UNDRAINED set to true. This decouples U and Pw and keeps the Pw hydrostatic is this stage
Then in stage 2 the following is used:
- in the material switched IGNORE_UNDRAINED to false. This couples U and Pw, excess pore pressure can now build up.
- keep the gravity load
- keep a boundary condition for the pressure ( = your Lower_Drainage_Fluid_Pressure boundary).
- still the side boundaries are used, but now as vertical support i.s.o. horizontal

All of this still gives the phenomenon that you have observed: the bottom moves, but the rest of the column stays horizontally in the same place. Searching in the code reveals that in line 194 the function  ApplyConstantScalarValueProcess::InternalApplyValue fixes are added to nodal variables, but these nodal_variables are not given free. This implies that fixes are only added, never released. 

When I free the nodal variables, the column moves horizontally. I will discuss in the coming days what the proper way is to fix this issue. 

Regards, Wijtze Pieter

**Reply by muhammedfurkanyilmaz** [reply]
Dear @WPK4FEM,

It has been some while, and there has not been any return related to this problem. And now we have to proceed with this in our project. I'll be pleased If you share the corresponding updates.

Kind Regards 
Furkan

**Comment by mcgicjn2** [comment]
Hi Furkan (@muhammedfurkanyilmaz),

We are currently assessing your problem. However for our prioritization it would be nice to know if you are working on a branch (or fork) of our code and if you plan to push the code to the master at some point soon. Otherwise it will not be a priority to us. 

If however you do have a branch please let us know which one, so we can help with your progress and measure how much work it will be on our end when you eventually push it to the master.

As you will appreciate although we want to support all research and everyone's use of our code, it is costly to us, if there is no return of new features, etc. to the community and our support will have no return value to us, thus at this moment we will only address the problem when we have time.

Regards

Jon (@mcgicjn2)

**Comment by muhammedfurkanyilmaz** [comment]
Hi Jon @mcgicjn2 ,

We have forked Kratos on the @ITUEQTEAM GitHub account to facilitate the implementation of our research. You may receive notifications as we update the repository. We deeply appreciate your assistance and support. We kindly request that you address the above-mentioned issue, enabling us to verify our findings on Kratos and advance our research.

Regards
Furkan

**Comment by rfaasse** [comment]
Hi Furkan @muhammedfurkanyilmaz,

Thank you for letting us know! We have looked into the issue and found a fix, after which the walls are no longer fixed in the second stage. You can find how implement it using the description in #12454. However, since there are ongoing developments regarding multistage analysis, we'll have to discuss (also with CIMNE) if this is the way we will fix the issue on master. Therefore, it might take longer for it to get there. 

In the meantime feel free to implement the described fix (a few additional lines in `geomechanics_analysis.py`) on the forked repo and let me or anyone from the team know if you still run into issues.

Kind regards,
Richard

**Comment by muhammedfurkanyilmaz** [comment]
Dear @rfaasse,

I've added the code snippet you shared for this problem, now the lateral boundaries can move freely laterally but input acceleration is not integrated correctly, the previously solved problem ( #12178 ) popped up again unless I use the displacement trace as an input. But I'm still getting high responses in terms of stress and strains compared to Plaxis even though I used the absorbing boundaries with relatively high damping.

Regards
Furkan

**Reply by rfaasse** [reply]
Hi @muhammedfurkanyilmaz,

Regarding your first issue: when checking e.g. the middle bottom boundary node and compare integrated displacement with the output displacement by kratos, they are in agreement (when looking at your earlier submitted [LysmerBoundaries-K0.zip](https://github.com/KratosMultiphysics/Kratos/files/15215518/LysmerBoundaries-K0.zip)) :

![image](https://github.com/user-attachments/assets/df71e4bf-cff7-4cfb-88ed-4d0dd8e6040c)

However, note that I needed to indicate the initial velocity in that node, since if I kept the initial velocity on 0, we'd get something like:
![image](https://github.com/user-attachments/assets/9ddf10ec-1691-43fd-91e9-f032fb3d352f)

I've attached the excel used to generate these plots [here](https://github.com/user-attachments/files/16659579/acceleration2displacement.xlsx) for your info.

Regarding the high stress/strain responses, we don't see any obvious problems in the input/output, could you provide a more specific example of what the expected and actual behavior is (e.g. which outputs, which locations and why), such that we can have a more targeted look at your problem?

Kind regards,
Richard

FYI @WPK4FEM @mcgicjn2
